support/rag/qa-webservice/private_ai_api.py

```diff
@@ -0,0 +1,109 @@
+from fastapi import FastAPI
+from pydantic import BaseModel
+import os
+import nltk
+import psycopg2
+from nltk.corpus import stopwords
+from nltk.stem import WordNetLemmatizer
+from sentence_transformers import SentenceTransformer
+import configparser
+from pgvector.psycopg2 import register_vector
+import re
+import requests
+import json
+from openai import OpenAI
+
+# TODO: figure out how to make the parts configurable, i.e. embedding model could be configured here
+# but it would also need to be the same at the document ingestion step so that the similarity search
+# can work
+
+
+def clean_text(text):
```

Comment:
Please, move all the reused logic about chunking and embedding in separate file. So it's clearly delianated. 

This would make it easier to reason about it and think how to re-use it properly (without copy pasting)